### DISCUSSION TITLE
Github Actions failing to run build (Vue web app)

### DISCUSSION THREAD

**Discussion by alexpop-ads** [community/community#69293] [Actions] [answered]
### Select Topic Area

Question

### Body

Hello. Long story short 2 weeks ago everything worked fine with Github actions. Now we only commited a small style change (a new CSS rule for an ID) and the build for CI/CD failed. We are deploying to an Azure environment. npm install works just fine, no other modification were done and last build was successful. This build however, during npm run build, Github gives us the following:
> vue-cli-service build

Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
All browser targets in the browserslist configuration have supported ES module.
Therefore we don't build two separate bundles for differential loading.

-  Building for production...
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme
undefined:1
{"type":"job","id":20,"error":null,"result":{"result":[{"buffer":true,"string":true},{"data":{"version":3,"names":["getCurrentInstance","inject","onUnmounted","onDeactivated","onActivated","computed","unref","watchEffect","defineComponent","reactive","h","provide","ref","watch","shallowRef","nextTick","setupDevtoolsPlugin","isBrowser","window","isESModule","obj","__esModule","Symbol","toStringTag","assign",.......................[THERE'S A HUGE AMOUNT OF CHARACTERS HERE]..... if (originalMatcher !== matcher)\r\n                    originalMatcher.alias.push(matcher);\r\n                // remove the route if named and only for the top record (avoid in nested calls)\r\n                // this works because the original record is the first one\r\n                if (isRootAdd && record.name && !isAliasRecord(matcher))\r\n                    removeRoute(record.name);\r\n            }\r\n            if (mainNormalizedRecord.children) {\r\n                const children = mainNo
Node.js v18.18.0
Error: Process completed with exit code 1.

Sometimes it may also say:
SyntaxError: Unexpected token  in JSON at position 219248
    at JSON.parse (<anonymous>)
    at /home/runner/work/app-frontend/app-frontend/node_modules/@vue/cli-plugin-babel/node_modules/thread-loader/dist/WorkerPool.js:144:30
    at Socket.onChunk (/home/runner/work/app-frontend/app-frontend/node_modules/@vue/cli-plugin-babel/node_modules/thread-loader/dist/readBuffer.js:[40](https://github.com/Org/app-frontend/actions/runs/6416326758/job/17419842856#step:4:41):9)
    at Socket.emit (node:events:517:28)
    at addChunk (node:internal/streams/readable:335:12)
    at readableAddChunk (node:internal/streams/readable:308:9)
    at Readable.push (node:internal/streams/readable:2[45](https://github.com/Org/app-frontend/actions/runs/6416326758/job/17419842856#step:4:46):10)
    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)

Node.js v18.18.0
Error: Process completed with exit code 1.
Here's our packages.json:
{
  "name": "App",
  "version": "0.2.0",
  "private": true,
  "scripts": {
    "start": "npm run serve",
    "serve": "vue-cli-service serve",
    "build": "vue-cli-service build",
    "lint": "vue-cli-service lint"
  },
  "dependencies": {
    "ag-grid-community": "^29.1.0",
    "ag-grid-vue3": "^29.1.0",
    "axios": "^1.3.4",
    "bulma": "^0.9.4",
    "chart.js": "^4.3.0",
    "core-js": "^3.8.3",
    "vue": "^3.2.13",
    "vue-chartjs": "^5.2.0",
    "vue-router": "^4.1.6"
  },
  "devDependencies": {
    "@babel/core": "^7.12.16",
    "@babel/eslint-parser": "^7.12.16",
    "@vue/cli-plugin-babel": "~5.0.0",
    "@vue/cli-plugin-eslint": "~5.0.0",
    "@vue/cli-service": "~5.0.0",
    "eslint": "^7.32.0",
    "eslint-plugin-vue": "^8.0.3"
  },
  "eslintConfig": {
    "root": true,
    "env": {
      "node": true
    },
    "extends": [
      "plugin:vue/vue3-essential",
      "eslint:recommended"
    ],
    "parserOptions": {
      "parser": "@babel/eslint-parser"
    },
    "rules": {
      "vue/no-unused-components": "off",
      "no-unudes-vars": "off"
    }
  },
  "browserslist": [
    "> 1%",
    "last 2 versions",
    "not dead",
    "not ie 11"
  ]
}

We tried everything from updating all packages to using the known working versions, checking the code etc. On our local machines it's working well. Any idea of how to solve this would be welcomed.

**Comment by waveoffire** [comment]
same problem here, tested with npm and yarn

**Comment by Laritello** [comment, marked as answer]
Same problem. Looks like it's related to this issue https://github.com/webpack-contrib/thread-loader/issues/191.

**Reply by alexpop-ads** [reply]
Thank you so much. I found this answer https://github.com/webpack-contrib/thread-loader/issues/191#issuecomment-1728002022 which was the solution. Used an environment variable to fix the issue during building and no more errors now :)

**Comment by MadhushaPrasad** [comment]


**Reply by Akash1134** [reply]
Hi @MadhushaPrasad , 

We’ve clarified our stance on using generative AI tools like ChatGPT within our Community via [this announcement](https://github.com/orgs/community/discussions/69006). Please review the guidelines to ensure your post meets them as failure to adhere to those rules can result in action taken by our moderator team.  You can read our updated [Code of Conduct](https://github.com/community/community/blob/baa5cd9020ef3f2bbf86dca1191d9c90f9b0822d/CODE_OF_CONDUCT.md#reasonable-use-of-ai-generated-content) and the [announcement](https://github.com/orgs/community/discussions/69006) for more details. Thank you for helping us maintain an authentic and beneficial space for everyone.
